File: Program/ActEditor/bin/pl/wroc/pwr/student/acteditor/model/ElementRegistry.js
Clazz.declarePackage ("pl.wroc.pwr.student.acteditor.model");
Clazz.load (null, "pl.wroc.pwr.student.acteditor.model.ElementRegistry", ["java.util.ArrayList", "$.Collections"], function () {
c$ = Clazz.decorateAsClass (function () {
this.elements = null;
Clazz.instantialize (this, arguments);
}, pl.wroc.pwr.student.acteditor.model, "ElementRegistry");
Clazz.makeConstructor (c$, 
($fz = function () {
this.elements = java.util.Collections.synchronizedList ( new java.util.ArrayList ());
}, $fz.isPrivate = true, $fz));
c$.getRegistry = Clazz.defineMethod (c$, "getRegistry", 
function () {
if (pl.wroc.pwr.student.acteditor.model.ElementRegistry.registry == null) {
($t$ = pl.wroc.pwr.student.acteditor.model.ElementRegistry.registry =  new pl.wroc.pwr.student.acteditor.model.ElementRegistry (), pl.wroc.pwr.student.acteditor.model.ElementRegistry.prototype.registry = pl.wroc.pwr.student.acteditor.model.ElementRegistry.registry, $t$);
}return pl.wroc.pwr.student.acteditor.model.ElementRegistry.registry;
});
Clazz.defineMethod (c$, "add", 
function (element) {
for (var each, $each = this.elements.iterator (); $each.hasNext () && ((each = $each.next ()) || true);) {
if ((each).getName ().equals (element.getName ())) {
return ;
}}
this.elements.add (element);
}, "pl.wroc.pwr.student.acteditor.parsing.xsd.Element");
Clazz.defineMethod (c$, "get", 
function (name) {
for (var each, $each = this.elements.iterator (); $each.hasNext () && ((each = $each.next ()) || true);) {
if ((each).getName ().equals (name)) {
return each;
}}
return null;
}, "~S");
Clazz.defineMethod (c$, "getElements", 
function () {
return this.elements;
});
Clazz.defineStatics (c$,
"registry", null);
});

File: Program/ActEditor/src/pl/wroc/pwr/student/acteditor/model/ElementRegistry.java
package pl.wroc.pwr.student.acteditor.model;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

import pl.wroc.pwr.student.acteditor.model.tags.Element;

/**
 * Zapewnia dostep do rejestru elementow.
 * 
 * Rejestr zawiera elementy Schematu XML, ktore sa identyfikowane po nazwie
 * elementu.
 * 
 * @author Mateusz
 * 
 */
public class ElementRegistry {
	private static ElementRegistry registry;
	private List elements;

	private ElementRegistry() {
		elements = Collections.synchronizedList(new ArrayList());
	}

	/**
	 * Zwraca referencje do rejestru elementów.
	 * 
	 * @return Referencja do rejestru elementów.
	 */
	public static ElementRegistry getRegistry() {
		if (registry == null) {
			registry = new ElementRegistry();
		}
		return registry;
	}

	/**
	 * Dodaje element do rejestru, jesli element o takiej nazwie nie istnieje
	 * jeszcze w rejestrze.
	 * 
	 * @param element
	 *          Element, który ma zostac dodany do rejestru.
	 */
	public void add(Element element) {
		for (Object each : elements) {
			if (((Element) each).getName().equals(element.getName())) {
				return;
			}
		}
		elements.add(element);
	}

	/**
	 * Zwraca element z rejestru o podanej nazwie.
	 * 
	 * @param name
	 *          Nazwa atrybutu w rejestrze.
	 * @return Element o podanej nazwie.
	 */
	public Element get(String name) {
		for (Object each : elements) {
			if (((Element) each).getName().equals(name)) {
				return (Element) each;
			}
		}
		return null;
	}

	/**
	 * Zwraca liste elementów istniejacych w rejestrze.
	 * 
	 * @return Lista elementów.
	 */
	public List getElements() {
		return elements;
	}
}
